mailpoet/assets/js/src/wizard/welcome_wizard_controller.tsx

```diff
@@ -71,31 +81,29 @@ function WelcomeWizardStepsController(props) {
   );
 
   const updateSender = useCallback(
-    (data) => {
+    (data: { address: string }) => {
       setSender({ ...sender, ...data });
     },
     [sender],
   );
 
   const submitSender = useCallback(() => {
-    updateSettings(createSenderSettings(sender)).then(() => redirect(step));
+    void updateSettings(createSenderSettings(sender)).then(() =>
+      redirect(step),
+    );
   }, [redirect, sender, step]);
 
   const skipSenderStep = useCallback(
     (e) => {
       e.preventDefault();
       setLoading(true);
-      updateSettings(
+      void updateSettings(
```

Comment:
Not sure what value does this `void` operator add in these case, ts seems like to be happy without it as well, and I guess removing it doesn't change the logic either.

Reply:
Without the `void` ESLint returns the following error:

```
/home/rodrigo/devel/mailpoet/mailpoet/assets/js/src/wizard/welcome_wizard_controller.tsx
  63:5  error  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
```

My assumption is that since we are already handling errors inside `updateSettings()`, it is safe to use `void` here instead of using `catch()` or the second parameter of `then()`, but I'm not 100% sure. What do you think?

Reply:
I'd prefer prefixing with `await` to better expose the exceptions if they happen specially since as a user we don't know if and how the exceptions are handled inside `updateSettings`, besides this function `submitTracking` is stalling the application by setting `setLoading(true);` so by nature it's sync and it's fine to await it.

Reply:
I don't know if I'm missing something as I'm not super familiar the all the concepts here, but I don't think I can simply use `await` as it should be used only inside an async function or at the top level of a module. Neither is the case for the places where I used `void`. Happy to chat synchronously about this tomorrow if it is easier so that we can find a solution.

Reply:
Thanks for the information you shared on Slack, Sam. This PR is ready for a final review.